src/components/manifold-subscription-create/manifold-subscription-create.tsx

```diff
@@ -205,36 +215,45 @@ export class ManifoldSubscriptionCreate {
         },
       });
 
-      this.subscribing = false;
-
       if (error) {
+        this.subscribing = false;
         this.setupIntentError = error.message;
-      } else {
-        this.setupIntentStatus = setupIntent?.status;
-        if (setupIntent?.status === 'succeeded') {
-          const configuredFeatures = Object.keys(this.configuredFeatures).map(key => ({
-            label: key,
-            value: `${this.configuredFeatures[key]}`,
-          }));
-
-          const variables: CreateSubscriptionMutationVariables = {
-            ownerId: undefined,
-            planId: this.planId,
-            configuredFeatures,
-          };
-
-          const res = await this.connection?.graphqlFetch<CreateSubscriptionMutation>({
-            query: createSubscrptionMutation,
-            variables,
-          });
-
-          // eslint-disable-next-line no-console
-          console.log({ res });
+        return;
+      }
+
+      this.setupIntentStatus = setupIntent?.status;
+      if (setupIntent?.status === 'succeeded' && this.connection) {
+        const configuredFeatures = Object.keys(this.configuredFeatures).map(key => ({
+          label: key,
+          value: `${this.configuredFeatures[key]}`,
+        }));
+
+        const variables: CreateSubscriptionMutationVariables = {
+          ownerId: undefined,
+          planId: this.planId,
+          configuredFeatures,
+        };
+
+        const { data, errors } = await this.connection.graphqlFetch<CreateSubscriptionMutation>({
+          query: createSubscrptionMutation,
+          variables,
+        });
+
+        if (errors) {
+          // TODO handle errors
+        }
+
+        if (data) {
+          console.log(data);
+          this.createSuccess.emit({ id: data.createSubscription.data.id });
         }
       }
+      this.subscribing = false;
     }
   };
 
+  @Event({ eventName: 'success' }) createSuccess: EventEmitter<{ id: string }>;
```

Comment:
might need a specific flag here to expose the event

Reply:
The event fires for me

```
    <script>
      const create = document.querySelector('manifold-subscription-create');
      create.addEventListener('success', console.log);
    </script>
```